scripts/asm-installer/install_asm

```diff
@@ -1977,6 +1982,16 @@ EOF
   outro
 }
 
+establish_managed_logging_bucket() {
+  gcloud beta logging buckets create ${LOGS_BUCKET} --project=${PROJECT_ID} --location=${CLUSTER_LOCATION}
```

Comment:
Does this require specific IAM permissions to do? What happens if you run this command and the bucket already exists?

Reply:
It requires: `logging.buckets.create`.

If you run the command again, you get an error: `ERROR: (gcloud.beta.logging.buckets.create) ALREADY_EXISTS: Bucket asm-control-plane-logs already exists`.

Reply:
What is the preferred way to handle non-idempotent operations in the install script?

Reply:
Check first and exit if it exists already. See the register_cluster() function for an example.

Could you the appropriate role to the required_iam_roles() function? Ref. https://cloud.google.com/iam/docs/understanding-roles#logging-roles

Reply:
Added a check.

Reply:
Updated required roles, validated check.